src/sidebar.cpp

```diff
@@ -0,0 +1,573 @@
+        veh = g->m.veh_at( pos() );
+    }
+    if( veh ) {
+        veh->print_fuel_indicators( w, sideStyle ? 2 : 3, sideStyle ? getmaxx( w ) - 5 : 49 );
+        nc_color col_indf1 = c_light_gray;
+
+        float strain = veh->strain();
+        nc_color col_vel = strain <= 0 ? c_light_blue :
+                           ( strain <= 0.2 ? c_yellow :
+                             ( strain <= 0.4 ? c_light_red : c_red ) );
+
+        // Draw the speedometer.
+        int speedox = sideStyle ? 0 : 28;
+        int speedoy = sideStyle ? 5 :  3;
+
+        bool metric = get_option<std::string>( "USE_METRIC_SPEEDS" ) == "km/h";
+        // Logic below is not applicable to translated units and should be changed
+        int velx    = metric ? 4 : 3; // strlen(units) + 1
+        int cruisex = metric ? 9 : 8; // strlen(units) + 6
+
+        if( !sideStyle ) {
+            if( !veh->cruise_on ) {
+                speedox += 2;
+            }
+            if( !metric ) {
+                speedox++;
+            }
+        }
+
+        const char *speedo = veh->cruise_on ? "%s....>...." : "%s....";
+        mvwprintz( w, speedoy, speedox,        col_indf1, speedo, velocity_units( VU_VEHICLE ) );
+        mvwprintz( w, speedoy, speedox + velx, col_vel,   "%4d",
+                   int( convert_velocity( veh->velocity, VU_VEHICLE ) ) );
+        if( veh->cruise_on ) {
+            mvwprintz( w, speedoy, speedox + cruisex, c_light_green, "%4d",
+                       int( convert_velocity( veh->cruise_velocity, VU_VEHICLE ) ) );
+        }
+
+        const int vel_offset = 11 + ( veh->velocity != 0 ? 2 : 0 );
+        wmove( w, sideStyle ? 4 : 3, getmaxx( w ) - vel_offset );
+        if( veh->velocity != 0 ) {
+            nc_color col_indc = veh->skidding ? c_red : c_green;
+            int dfm = veh->face.dir() - veh->move.dir();
+
+            if( dfm == 0 ) {
+                wprintz( w, col_indc, "^" );
+            } else if( dfm < 0 ) {
+                wprintz( w, col_indc, "<" );
+            } else {
+                wprintz( w, col_indc, ">" );
+            }
+
+            wprintz( w, c_white, " " );
+        }
+
+        //Vehicle direction indicator in 0-359° where 0 is north (veh->face.dir() 0° is west)
+
+        wprintz( w, c_white, string_format( "%3d°", ( veh->face.dir() + 90 ) % 360 ).c_str() );
+
+        if( sideStyle ) {
+            // Make sure this is left-aligned.
+            mvwprintz( w, speedoy, getmaxx( w ) - 9, c_white, "%s", _( "Stm " ) );
+            print_stamina_bar( *this, w );
+        }
+    } else {  // Not in vehicle
+        nc_color col_str = c_white, col_dex = c_white, col_int = c_white,
+                 col_per = c_white, col_spd = c_white, col_time = c_white;
+        int str_bonus = get_str_bonus();
+        int dex_bonus = get_dex_bonus();
+        int int_bonus = get_int_bonus();
+        int per_bonus = get_per_bonus();
+        int spd_bonus = get_speed_bonus();
+        if( str_bonus < 0 ) {
```

Comment:
Lot's of repeated code (for each stat) that could be put into a function.

Reply:
Currently it's a simple 'copy-paste' PR, I'd prefer to keep further improvements for other PRs.

Reply:
:+1: